parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -262,12 +313,17 @@ public void writeTo(OutputStream out) throws IOException {
     }
 
     @Override
+    public byte[] getBytesUnsafe() {
+      return getBytes();
```

Comment:
Caching will take memory. It makes sense to cache it only if it is being called again and again. If it is being called again and again, then the consumer must take care of caching. That's my understanding. Let me know if you disagree.